Support Debian testing distros

conorsch opened the pull request:
The build role already supports stable distros for Debian and Ubuntu, but with a few tweaks, we can also use it for local compilation on a developer workstation running Debian Stretch/Sid (current Testing/stable).

The distro-specific vars files are a bit repetitious, but making them DRY would involve breaking the apt packages up into several lists, which actually makes it harder to understand what's being installed. The distro-specific vars files also provide cheap distro validation:

```
TASK [build-grsec-kernel : Import OS-specific dependency list.] ****************
task path: /home/conor/freedomofpress/ansible-role-grsecurity/roles/build-grsec-kernel/tasks/packages.yml:12
fatal: [grsec-build]: FAILED! => {"failed": true, "msg": "No file was found when using with_first_found. Use the 'skip: true' option to allow this task to be skipped if no files are found"}
        to retry, use: --limit @examples/build-grsecurity-kernel-test.retry
```

It would probably be better to catch that error via Ansible 2's `block` syntax and display a more helpful message.

conorsch committed:
Renames distro-specific vars files

Now including both the distro and release, e.g. "Debian_jessie" rather
than simply "Debian." Opens the door for providing more targeted distro
support.

conorsch committed:
Adds vars file for Debian stretch/sid

Ansible populates the `ansible_distribution_release` with "NA" for
Stretch or Sid, so let's iterate over a couple vars file name formats
via `with_first_found` and check for the distribution version, which
oddly will be "stretch/sid" rather than a numeral.

conorsch committed:
Adds specific vars file for Ubuntu Trusty

Needs an older version of the gcc packages, 4.8 rather than 4.9 in
Vivid.

conorsch committed:
Adds python-requests to apt packages for build role

The custom facts module that ships with the build role requires
python-requests to be installed, so let's pull it in as part of the apt
config. It was already automatically installed on Ubuntu hosts.

conorsch committed:
Namespaces the apt packages var in build role

Adding prefix `grsecurity_build_` as with the other role vars to prevent
accidental clobbering in site configs. Also updated the interpolation of
the list var in the install task, to resolve a deprecation warning since
Ansible v2.

conorsch committed:
Only installs git if ubuntu overlay is enabled

The only tasks that require git are skipped by default, so tie the git
install logic to the ubuntu-overlay logic, and install git within the
ubuntu-overlay task list.

conorsch commented:
> It would probably be better to catch that error via Ansible 2's `block` syntax and display a more helpful message.

Doing so would close #33, so I'll tack that on here since it's relevant to the distro vars changes.

msheiny commented:
I like the approach you took to abstracting the distros here -- I'm not sure you really need a useful in-line error message though. I always find those hard to read in the ansible output and I think it's pretty obvious when you get the error and after you glance at the task what is going on. I think it make more sense to throw a message in the readme explaining distro support and how one might start to add newer ones. Though honestly I didn't realize ansible added the `block` structure in `2.0` so if we just want an excuse to play with it - I say go wild :smiley:

conorsch committed:
Display informative error message if distro unsupported

Thanks to Ansible v2's `block` syntax, we can catch an error when
importing apt packages vars, and display an informative message about
the specific distribution that's unsupported, as well as solicit PRs to
add support.

conorsch commented:
The error-handling via `block` makes for a very nice failure scenario. Attempting to apply to a Debian Wheezy host:

```
TASK [build-grsec-kernel : Import OS-specific dependency list.] ****************
task path: 
/home/conor/freedomofpress/ansible-role-grsecurity/roles/build-grsec-kernel/tasks/packages.yml:6
fatal: [grsec-build]: FAILED! => {"failed": true, "msg": "No file was found when using 
with_first_found. Use the 'skip: true' option to allow this task to be skipped if no files are 
found"}

TASK [build-grsec-kernel : Validate support for compiling kernel on current OS.] ***
task path: 
/home/conor/freedomofpress/ansible-role-grsecurity/roles/build-grsec-kernel/tasks/packages.yml:12
fatal: [grsec-build]: FAILED! => {"changed": false, "failed": true, "msg": "Cannot compile a 
grsecurity-patched kernel under Debian wheezy. The problem may be as simple as a few missing 
package dependencies. Send a pull request to add support for your distro! 
https://github.com/freedomofpress/ansible-role-grsecurity"}
        to retry, use: --limit @examples/build-grsecurity-kernel-test.retry

PLAY RECAP *********************************************************************
grsec-build                : ok=2    changed=0    unreachable=0    failed=1

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

Big fan of that!
